crl/storer/storer.go

```diff
@@ -151,9 +159,47 @@ func (cs *crlStorer) UploadCRL(stream cspb.CRLStorer_UploadCRLServer) error {
 		return fmt.Errorf("validating signature for %s: %w", crlId, err)
 	}
 
+	// Before uploading this CRL, we want to compare it against the previous CRL
+	// to ensure that the CRL Number field is not going backwards. This is an
+	// additional safety check against clock skew and potential races, if multiple
+	// crl-updaters are working on the same shard at the same time. We only run
+	// these checks if we found a CRL, so we don't block uploading brand new CRLs.
+	filename := fmt.Sprintf("%d/%d.crl", issuer.NameID(), shardIdx)
+	prevObj, err := cs.s3Client.GetObject(stream.Context(), &s3.GetObjectInput{
+		Bucket: &cs.s3Bucket,
+		Key:    &filename,
+	})
+	if err != nil {
+		smithyErr := &smithyhttp.ResponseError{}
```

Comment:
Nit: I would say `var smithyErr *smithyhttp.ResponseError`. That makes it clearer this value isn't used, since the pointer starts as nil. Also it's a little less punctuation.